Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -193,6 +193,14 @@ extension NSManagedObjectContext {
         try payload.messages.forEach { _ = try saveMessage(payload: $0, for: payload.channel.cid) }
 
         dto.updateOldestMessageAt(payload: payload)
+        
+        // Backend only returns a boolean for hidden state on channel query
+        // If backend returns true, we know channel is hidden
+        // but don't know when. Our best guess is that channel is hidden when
+        // query is made
+        if payload.hidden == true {
+            dto.hiddenAt = Date()
```

Comment:
same thing, I suggest picking the date from server-side data and not local time

Reply:
done